### PR TITLE
feat(components): add base for tab component

### PR DESCRIPTION
**Pull request by ejirocodes**
This is a WIP that addresses tabs component #23

### CODE REVIEW

**Comment by productdevbook** [comment]
@ejirocodes sorry, @dammy001  accidentally posted a merge commit here

**Comment by productdevbook** [comment]
What is the situation with this issue, if you want, I can support you.

**Comment by ejirocodes** [comment]
@productdevbook, working on it tonight. Is there a discord channel I can reach you on?

**Comment by productdevbook** [comment]
Yes, [discord](chat.productdevbook.com)

**Comment by productdevbook** [comment]
Yes, check website or this link [discord](https://chat.productdevbook.com)

**Comment by ejirocodes** [comment]
> Yes, check website or this link [discord](https://chat.productdevbook.com)

@productdevbook, seems like the invite expired 
image

**Comment by productdevbook** [comment]
https://chat.productdevbook.com

**Comment by ejirocodes** [comment]
> chat.productdevbook.com

Same thing @productdevbook. Please consider regenerating the link. Maybe it has expired or something  
image

**Comment by productdevbook** [comment]
https://discord.gg/dpUJmRmugD

**Comment by ejirocodes** [comment]
> [discord.gg/dpUJmRmugD](https://discord.gg/dpUJmRmugD)

@productdevbook, still not working man

**Comment by dammy001** [comment]
@ejirocodes can you try this: https://discord.gg/9TFgjXvUcb

**Comment by ejirocodes** [comment]
@productdevbook, this has been implemented. If you could please take a look and get the storybook to work properly. LMK if you have any questions.
Will write unit test for this component once it's finialised

File: packages/core/use-composable/src/index.ts [hunk 1]
```diff
@@ -10,3 +10,5 @@ export { useId } from './useId'
 export type { MaybeComputedElementRef } from './unrefElement'
 export { usePrimitiveElement } from './usePrimitiveElement'
 export { useArrowNavigation } from './useArrowNavigation'
+export { computedEager, syncRef }
+\
```
Comment (productdevbook) [inline packages/core/use-composable/src/index.ts]: same merged problem, i am now working

**Comment by productdevbook** [comment]
> @productdevbook @ejirocodes Component file naming convention should follow camelCase and not kebab-case

Yes, but not now, the last. Before publishing. Please do not change file names at the moment. Because we can understand what is done.

**Comment by productdevbook** [comment]
@ejirocodes  Hi, `useRef` and more change, can you take a look at the new structure and other components and update your codes here?

**Comment by productdevbook** [comment]
i'm working on it now

**Review by productdevbook — APPROVED** [review]

**Comment by productdevbook** [comment]
@ejirocodes thank, `write unit test for this component` you do this in the new prda. We will be waiting.

**Pull request merged**